packages/flutter_driver/lib/src/timeline_summary.dart

```diff
@@ -0,0 +1,132 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert' show JSON, JsonEncoder;
+
+import 'package:file/file.dart';
+import 'package:path/path.dart' as path;
+
+import 'common.dart';
+
+const String _kDefaultDirectory = 'build';
+final JsonEncoder _prettyEncoder = new JsonEncoder.withIndent('  ');
+
+/// Extracts statistics from the event loop timeline.
+TimelineSummary summarizeTimeline(Map<String, dynamic> timeline) {
+  return new TimelineSummary(timeline);
+}
+
+class TimelineSummary {
+  TimelineSummary(this._timeline);
+
+  final Map<String, dynamic> _timeline;
+
+  /// Average amount of time spent per frame in the framework building widgets,
+  /// updating layout, painting and compositing.
+  double get averageFrameBuildTimeMillis {
+    int totalBuildTimeMicros = 0;
+    int frameCount = 0;
+
+    for (TimedEvent event in _extractBeginFrameEvents()) {
+      frameCount++;
+      totalBuildTimeMicros += event.duration.inMicroseconds;
+    }
+
+    return frameCount > 0
+      ? totalBuildTimeMicros / frameCount / 1000
+      : null;
+  }
+
+  /// The total number of frames recorded in the timeline.
+  int get frameCount => _extractBeginFrameEvents().length;
+
+  /// The maximum amount of time considered safe to spend for a frame's build
+  /// phase. Anything past that is in the danger of missing the frame as 60FPS.
+  static const Duration kBuildBudget = const Duration(milliseconds: 8);
+
+  /// The number of frames that missed the build budget and therefore are in the
+  /// danger of missing frames.
+  ///
+  /// See [kBuildBudget].
+  int get missedFrameBuildBudgetCount => _extractBeginFrameEvents()
```

Comment:
(but also converted the API from getters to methods, because getters don't take args)